full-stack-tests/backend/src/integration/CloudSqlite.test.ts

```diff
@@ -188,6 +191,90 @@ describe("CloudSqlite", () => {
     container.disconnect({detach: true});
   });
 
+  it("Should upload a single user's changes successfully when a write lock is expired", async () => {
+    const testContainer1 = testContainers[0];
+    const testCache = azSqlite.makeCaches(["testCache"])[0];
+    testContainer1.connect(testCache);
+    testContainer1.acquireWriteLock(user1);
+    await testContainer1.copyDatabase("testBim", "testBimCopy");
+    const db = await BriefcaseDb.open({ fileName: "testBimCopy", container: testContainer1 });
+    db.saveFileProperty({ name: "logMask", namespace: "logMaskTest", id: 1, subId: 1 }, "this is a test");
+    db.close();
+    // set the expires time to five mins earlier, which avoids waiting for 5 mins here.
+    await azSqlite.subtractFromCurrentWriteLockExpiryTime(testContainer1, "bcv_kv.bcv", 5);
+    await expect(testContainer1.uploadChanges()).to.eventually.be.fulfilled;
+    testContainer1.disconnect();
+  });
+
+  it("Should user1 fail to upload changes if user2 holds write lock after user1's expiration time", async () => {
+    // simulate two users in two processes
+    // test container 1 and test container 2 are actually the same cloud container
+    const testContainer1 = testContainers[3];
+    const testContainer2 = testContainers[4];
+    expect(testContainer1.containerId).equal(testContainer2.containerId);
+    expect(testContainer1.baseUri).equal(testContainer2.baseUri);
+    const testCache1 = azSqlite.makeCaches(["testCache1"])[0];
+    const testCache2 = azSqlite.makeCaches(["testCache2"])[0];
+    // user1 grabs the write lock, make some changes but don't release the write lock
+    // after a specific amount of time, the write lock expires
+    testContainer1.connect(testCache1);
+    testContainer1.acquireWriteLock(user1);
+    await testContainer1.copyDatabase("testBim", "testBimCopy1");
+    const db1 = await BriefcaseDb.open({ fileName: "testBimCopy1", container: testContainer1 });
+    db1.saveFileProperty({ name: "upload", namespace: "uploadTest", id: 1, subId: 1 }, "this is a test");
+    db1.close();
+    // set the expires time to five mins earlier, which avoids waiting for 5 mins here.
+    await azSqlite.subtractFromCurrentWriteLockExpiryTime(testContainer1, "bcv_kv.bcv", 5);
```

Comment:
Similar to above, youre subtracting 5 mins here, but isn't the writeLockExpireSeconds on the testContainers[3] and testContainers[4] in seconds? You should convert them to minutes up above by multiplying by 60.

Reply:
`m_lockExpireSeconds = std::min((int) (12*SECONDS_PER_HOUR), std::max(5*((int)SECONDS_PER_MINUTE), m_lockExpireSeconds));`

So for testContainers[3] and testContainers[4], the m_lockExpireSeconds is 5 mins.